pyiceberg/table/inspect.py

```diff
@@ -256,8 +305,16 @@ def refs(self) -> "pa.Table":
         return pa.Table.from_pylist(ref_results, schema=ref_schema)
 
     def partitions(self, snapshot_id: Optional[int] = None) -> "pa.Table":
-        import pyarrow as pa
+        """
+        Retrieve partition information from the Iceberg table as a PyArrow Table.
+
+        Args:
+            snapshot_id (Optional[int]): The snapshot ID to filter partitions. If not provided, all partitions are included.
```

Comment:
> If not provided, all partitions are included.

I dont think this is right, the `snapshot_id` for all of these functions are used to time travel to a specific snapshot. otherwise the current snapshot will be used